parquet/src/main/java/org/apache/iceberg/parquet/VectorizedParquetReader.java

```diff
@@ -141,8 +148,15 @@ public T next() {
         advance();
       }
 
+      long remainingValues = nextRowGroupStart - valuesRead;
+      int remainingLimit = (int) (pushedLimit - valuesRead);
```

Comment:
why do we have to cast here? Isn't it int - int?

Reply:
It also strange that we track remaining limit in int and remaining values in long, but probably not a real world concern

Reply:
Typically, limit will not be a large number, but I changed `remainingLimit` to a long to make it consistent with `remainingValues`.